src/Cli/dotnet/commands/dotnet-add/AddCommandParser.cs

```diff
@@ -1,26 +1,35 @@
 // Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using System;
-using System.Linq;
-using Microsoft.DotNet.Cli.CommandLine;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.CommandLine.Parsing;
 using Microsoft.DotNet.Tools;
+using Microsoft.DotNet.Tools.Add;
 using LocalizableStrings = Microsoft.DotNet.Tools.Add.LocalizableStrings;
 
 namespace Microsoft.DotNet.Cli
 {
     internal static class AddCommandParser
     {
-        public static Command Add() =>
-            Create.Command(
-                "add",
-                LocalizableStrings.NetAddCommand,
-                Accept.ExactlyOneArgument()
-                      .DefaultToCurrentDirectory()
-                      .With(name: CommonLocalizableStrings.ProjectArgumentName,
-                            description: CommonLocalizableStrings.ProjectArgumentDescription),
-                AddPackageParser.AddPackage(),
-                AddProjectToProjectReferenceParser.AddProjectReference(),
-                CommonOptions.HelpOption());
+        public static readonly Argument ProjectArgument = new Argument(CommonLocalizableStrings.ProjectArgumentName)
+        {
+            Description = CommonLocalizableStrings.ProjectArgumentDescription,
+            Arity = ArgumentArity.ExactlyOne
```

Comment:
Arity should be unnecessary if you use `Argument<T>`. (You might want to use `Argument<FileInfo>` here.)